Framework/PythonInterface/mantid/plots/__init__.py

```diff
@@ -1045,6 +1048,217 @@ def tricontourf(self, *args, **kwargs):
         """
         return self._plot_2d_func('tricontourf', *args, **kwargs)
 
+    def set_initial_dimensions(self):
```

Comment:
This has all been done well. My only concern with the code now is that we have added a lot of weight to the MantidAxes api. These functions can be called by users in scripts when they set the `mantid` matplotlib projection. but many look like "helpers"?
The ones I would consider good api methods would be:
* `is_waterfall` (I think the `_plot` is a little redundant)
* `update_waterfall` (again I think the `_plot` is a little redundant)
* `set_waterfall` (see comment below on convert to/from waterfall). Possibly with extra options for the x,y offset and if fill is enabled? There is then a single api call to set the properties of a waterfall? `update_waterfall` could then be private or a non-method helper?

I think the rest can be private or pushed to non-class helper functions that accept an Axes parameter?